lib/task/tools/settingsTask.class.php

```diff
@@ -0,0 +1,351 @@
+        if (!empty($options['from-file']) && !is_readable($options['from-file'])) {
+            throw new Exception("The 'from-file' option must refer to a readable file.");
+        }
+
+        // Check that the "to-file" option is being used for the appropriate operation
+        if (!empty($options['to-file']) && 'get' != $arguments['operation']) {
+            throw new Exception("The 'to-file' option must only be used with the 'get' operation.");
+        }
+
+        // Check that the file specified by the "to-file" option doesn't already exist
+        if (!empty($options['to-file']) && file_exists($options['to-file'])) {
+            throw new Exception("The 'to-file' option mustn't refer to an already existing file.");
+        }
+
+        // Check that the file specified by the "to-file" option can be written to
+        if (!empty($options['to-file'])) {
+            // Do test file write
+            $fp = fopen($options['to-file'], 'w');
+
+            if (false === $fp) {
+                throw new Exception("The 'to-file' option must refer to a writeable file path.");
+            }
+
+            // Clean up after test file write
+            fclose($fp);
+            unlink($options['to-file']);
+        }
+    }
+
+    /**
+     * @see sfBaseTask
+     */
+    protected function configure()
+    {
+        $this->addArguments([
+            new sfCommandArgument(
+                'operation',
+                sfCommandArgument::REQUIRED,
+                'Setting operation ("get" or "set").'
+            ),
+            new sfCommandArgument(
+                'name',
+                sfCommandArgument::REQUIRED,
+                'Name of setting.'
+            ),
+        ]);
+
+        $this->addOptions([
+            new sfCommandOption(
+                'application',
+                null,
+                sfCommandOption::PARAMETER_OPTIONAL,
+                'The application name',
+                'qubit'
+            ),
+            new sfCommandOption(
+                'env',
+                null,
+                sfCommandOption::PARAMETER_REQUIRED,
+                'The environment',
+                'cli'
+            ),
+            new sfCommandOption(
+                'connection',
+                null,
+                sfCommandOption::PARAMETER_REQUIRED,
+                'The connection name',
+                'propel'
+            ),
+
+            // Tool options
+            new sfCommandOption(
+                'culture',
+                null,
+                sfCommandOption::PARAMETER_OPTIONAL,
+                'Culture (default: "en"))',
+                'en'
+            ),
+            new sfCommandOption(
+                'scope',
+                null,
+                sfCommandOption::PARAMETER_OPTIONAL,
+                'Scope of setting to get or set',
+                null
+            ),
+            new sfCommandOption(
```

Comment:
Opinion: I think it's more intuitive to make the 'value' an optional argument, so the command can be called like `symfony tools:settings set [name] [value]`